### ISSUE TITLE
[Feature]: Enable TI training CSV and JSON dynamic naming

### ISSUE THREAD

**Issue by Woisek** [vladmandic/automatic#644]
### Issue Description

Just finished a TI training and files I usually check afterwards are missing.  In the settings, "Save an csv containing the loss to log directory every N steps, 0 to disable" is set to 1, but the "textual_inversion_loss.csv" was not written. Also, a JSON file with the settings at begin of the training (like i.e. settings-2023-04-24-23-31-55.json) is missing.

Any suggestions on that?

### Version Platform Description

Win10, Firefox 106
Version: 67c39202 Sun Apr 30 11:16:58 2023 +0900

**Comment by vladmandic** [comment]
- for CSV: settings -> training-> Embeddings train log file
- for JSON: train -> Log directory

**Comment by Woisek** [comment]
> for CSV: settings -> training-> Embeddings train log file

No such option.

![Zwischenablagebild](https://user-images.githubusercontent.com/122026301/235354407-60fea937-be14-4ff0-9c8d-ec7653408012.png)

for JSON: train -> Log directory

Not found here

![Zwischenablagebild (1)](https://user-images.githubusercontent.com/122026301/235354426-9c0d273b-b6a0-4488-8934-12af3c21cf64.png)

>btw, for questions, better to open discussion thread first and if there is issue, then we can work on it under issues.

Well, for me it seems like an issue, unless I miss something.

**Comment by vladmandic** [comment]
csv settings were moved from settings -> paths to settings -> train very recently, either update repo or find them in paths.
for json, setting is right there in the middle of your screenshot?

**Comment by Woisek** [comment]
Hold on ... in the Settings -> System paths, there is "Embeddings train log file" for the training loss csv file.
OK, but what do I have to enter here to get this file saved in the folder created when doing a training like it was always? This obviously can't be a fixed directory, otherwise the file would get overwritten with every new training. Why a path setting anyway? It should be saved, wherever the training folder is created.
And this doesn't answer the question of the additional date saved JSON settings file when a training begins.

**Comment by vladmandic** [comment]
as both are fixed right now (as configured in UI), this then becomes a feature request to make filenames dynamic - totally valid request, i'll see how fast i can do that. reopening and converting to feature request.

**Comment by Woisek** [comment]
Many thanks for your quick respond, but ... why don't you let it be as it was before? IMHO there is no need to have the User change the path to a detail like a log file (that is part of something bigger). Maybe the general path to where the embeddings are created, this is cool.
But everything else was good as it was. 🙂 I also miss the fact, that a date-time folder is created with the embedding name as folder inside, not just only a folder with the embedding name. Again, after a train failure and re-train with different settings but the same name, there is a chance to overwrite something and/or messing things up. I guess no one will think about renaming the old train folder before beginning a new training. 😁

**Comment by vladmandic** [comment]
the feature request is fully valid and i'll see how fast i can do it.
but saying "why don't you let it be as before" is not really as different people have different preferences.

**Comment by Woisek** [comment]
You're absolutely right about the different preferences of people ... so it's itching me to know, what the motivation was to change this part in it's working. 🙂

**Comment by vladmandic** [comment]
done, both are relative to log_directory.

```python
json = f"{params['embedding_name']}-{now.strftime('%Y-%m-%d_%H-%M-%S')}.json"
csv = f"{embedding_name}.csv"
```

**Comment by Woisek** [comment]
Cool! Will check on the next TI training. 👍
Thanks!
